### ISSUE TITLE
Support for numpy matrix operations

### ISSUE THREAD

**Issue by alexfeld** [hgrecco/pint#470]
A few things regarding matrix operations that aren't working properly that would be useful.

- support for `np.dot` e.g. `A.dot(b)` or `b.dot(A)`. Currently, if `A,b` are arrays containing quantities, this works. However, if `A,b` are `Quantity`'s, then this operation does not return the correct result. I believe this is related to #39 . 
- support for `np.linalg.solve`. Solving a system where `A,b` are instances of `Quantity` the method fails with a `TypeError` in line 1235 of `quantity`. I think this is related to #420. If `A,b` are arrays of `Quantity`s the method fails in `linalg` with a `TypeError: no loop matching the specified signature was found for ufunc solve1`

**Comment by awa5114** [comment]
Has this been implemented? The documentation states that `Pint` supports numpy array operations. A dot product is one of the most basic array operations. Otherwise I don't believe there is a point in supporting numpy....

**Comment by awa5114** [comment]
Hello? How difficult would an actual solution be to implement? I could try to submit a PR if its not too complicated. This functionality is very important!

**Comment by hgrecco** [comment]
Revisit after #905

**Comment by jthielen** [comment]
`np.dot` has been implemented in #905, but `np.linalg.solve` has not. I suspect it may be as simple as adding

```python
for func_str in ['solve']:
    implement_func('function', func_str, input_units=None, output_unit='div')
```

to `numpy_func.py`, but it may be complicated by the fact that `np.linalg.solve` is in a submodule.
